File: cases/taint/scala/array-element-positive/ArrayElementPositive.scala
package dataflowbench

object ArrayElementPositive {
  def dfb_source(): Int = { // DFB-SOURCE: array-element-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: array-element-sink

  def run(): Unit = {
    val values = new Array[Int](2)
    values(0) = dfb_source() // DFB-WITNESS: array-element-store
    values(1) = 0
    dfb_sink(values(0))
  }
}

File: cases/taint/typescript/array-element-positive/ArrayElementPositive.ts
function dfb_source(): string { // DFB-SOURCE: array-element-input
  return "tainted";
}

function dfb_sink(value: string): void {} // DFB-SINK: array-element-sink

function run(): void {
  const values: string[] = ["clean", "clean"];
  values[0] = dfb_source(); // DFB-WITNESS: array-element-store
  values[1] = "clean";
  dfb_sink(values[0]);
}
